pkg/s3wrapper/filesystem.go

```diff
@@ -331,7 +339,29 @@ func (f *FSClient) UploadBootFiles(ctx context.Context, openshiftVersion string)
 
 	isoFilePath := filepath.Join(f.basedir, isoObjectName)
 
-	return ExtractBootFilesFromISOAndUpload(ctx, log, isoFilePath, isoObjectName, rhcosImageURL, f)
+	if !baseExists {
+		if err = ExtractBootFilesFromISOAndUpload(ctx, log, isoFilePath, isoObjectName, rhcosImageURL, f); err != nil {
+			return err
+		}
+	}
+
+	if !minimalExists {
+		editor := isoeditor.CreateEditor(isoFilePath, openshiftVersion, log)
+
+		log.Infof("Extracting rhcos ISO (%s)", isoFilePath)
+		minimalIsoPath, err := editor.CreateMinimalISOTemplate(serviceBaseURL)
+		if err != nil {
+			log.Errorf("Error extracting rhcos ISO (%v)", err)
+			return err
+		}
+		defer os.Remove(minimalIsoPath)
+
+		// upload the minimal iso
+		log.Infof("Uploading minimal ISO (%s)", minimalIsoPath)
+		return f.UploadFileToPublicBucket(ctx, minimalIsoPath, minimalIsoObject)
```

Comment:
This looks like it could probably be pulled into a shared function so we're not duplicating so much.

Reply:
Actually tried to extract the entire function to util but it was a mess :) these specific lines should be simple enough.